gnucash/gnome/dialog-stock-editor.c

```diff
@@ -0,0 +1,927 @@
+
+#include "dialog-utils.h"
+#include "engine-helpers.h"
+#include "gnc-account-sel.h"
+#include "gnc-amount-edit.h"
+#include "gnc-date-edit.h"
+#include "gnc-component-manager.h"
+#include "gnc-event.h"
+#include "gnc-gnome-utils.h"
+#include "gnc-helpers.h"
+#include "gnc-main-window.h"
+#include "gnc-plugin-page-register.h"
+#include "gnc-session.h"
+#include "gnc-ui.h"
+#include "gnc-ui-balances.h"
+#include "dialog-stock-editor.h"
+
+static QofLogModule log_module = GNC_MOD_GUI;
+
+enum
+{
+     ACTION_COL_LABEL = 0,
+     ACTION_COL_ASSET_AMOUNT_MASK,
+     ACTION_COL_ASSET_VALUE_MASK,
+     ACTION_COL_PROCEEDS_MASK,
+     ACTION_COL_PROCEEDS,
+     ACTION_COL_DIVIDEND_MASK,
+     ACTION_COL_DIVIDEND,
+     ACTION_COL_CAPGAINS_MASK,
+     ACTION_COL_CAPGAINS,
+     ACTION_COL_EXPENSES_MASK,
+     ACTION_COL_EXPENSES,
+     ACTION_COL_NUM_COLUMNS
+};
+
+enum
+{
+     MASK_DISABLED = 0,
+     MASK_POSITIVE = 1,
+     MASK_ZERO     = 2,
+     MASK_NEGATIVE = 4,
+     MASK_CAPITALIZE = 8
+};
+
+/** STRUCTS *********************************************************/
+struct _StockEditorWindow
+{
+    Account *asset_account;        /* The stock account */
+    gnc_commodity *trans_currency;
+    time64 latest_split_date;
+    gint component_id;       /* id of component */
+
+    GtkWidget *window;       /* The stock-editor window                 */
+    GtkWidget *date_entry;
+    GtkWidget *action_combobox;
+
+    GtkWidget *current_balance_label;
+    GtkWidget *new_balance_label;
+    GtkWidget *price_label;
+
+    GtkWidget *warning_icon;
+    GtkWidget *warning_text;
+
+    GtkWidget *proceeds_acc;
+    GtkWidget *dividend_acc;
+    GtkWidget *capgains_acc;
+    GtkWidget *fees_acc;
+    GtkWidget *fees_capitalize;
+
+    GNCAmountEdit *asset_amount;
+    GNCAmountEdit *asset_basis;
+    GNCAmountEdit *proceeds_val;
+    GNCAmountEdit *dividend_val;
+    GNCAmountEdit *capgains_val;
+    GNCAmountEdit *sale_basis_val;
+    GNCAmountEdit *fees_val;
+
+    GtkWidget *asset_memo;
+    GtkWidget *proceeds_memo;
+    GtkWidget *dividend_memo;
+    GtkWidget *capgains_memo;
+    GtkWidget *fees_memo;
+    GtkWidget *description_entry;
+
+    GtkWidget *auto_capgain;
+
+    GtkWidget *ok_button;
+    GtkWidget *cancel_button;
+};
+
+typedef struct _StockEditorWindow StockEditorWindow;
+
+
+static void
+gnc_stockeditor_set_title_name (GtkWidget *window, Account *account)
+{
+    gchar *fullname = gnc_account_get_full_name (account);
+    gchar *title = g_strconcat (fullname, " - ", _("Stock Editor"), NULL);
+    gtk_window_set_title (GTK_WINDOW (window), title);
+    g_free (fullname);
+    g_free (title);
+}
+
+static void
+stock_editor_destroy (StockEditorWindow *data)
+{
+    gtk_widget_destroy (data->window);
+    g_free (data);
+}
+
+static void
+cancel_button_cb (GtkWidget *widget, StockEditorWindow *data)
+{
+    stock_editor_destroy (data);
+}
+
+static gboolean
+gae_unfocus (GtkWidget *widget, const StockEditorWindow *data)
```

Comment:
'gae' and 'gas' are a bit terse, how about e.g. `account_edit_unfocus` and `account_sel_get_account`?